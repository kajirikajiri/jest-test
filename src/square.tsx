import * as React from "react";

interface SquareProps {
  value: string;
}

interface SquareState {
  value: string;
}

export class Square extends React.Component<SquareProps, SquareState> {
  static defaultProps: SquareProps = {
    value: "",
  };

  constructor(props: SquareProps) {
    super(props);
    this.state = {
      value: "",
    };
  }

  render() {
    return (
      <button className="square" onClick={() => this.setState({ value: "X" })}>
        {this.state.value}
      </button>
    );
  }
}
