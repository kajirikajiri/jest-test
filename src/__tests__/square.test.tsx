import React from "react";
import { Square } from "../square";
import renderer from "react-test-renderer";

test("Square class", () => {
  const component = renderer.create(<Square onClick={() => {}} value={"1"} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
