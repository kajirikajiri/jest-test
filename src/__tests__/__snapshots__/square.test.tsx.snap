// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square class 1`] = `
<button
  className="square"
  onClick={[Function]}
>
  
</button>
`;
